src/CacheDecorator.php

```diff
@@ -96,13 +102,17 @@ private function uncompress(CacheItemInterface $item): CacheItemInterface
         if (!is_string($value)) {
             return $item;
         }
-        if ($value === null) {
-            return $item;
-        }
 
-        $item->set(
-            unserialize(gzuncompress($value))
-        );
+        // We try unserialize before uncompress because there are more not compressed cache entries
```

Comment:
Instead of having to try the unserialize call I'd prepend a little marker to the cached payload.

```
const COMPRESSION_MARKER = 'c:';

if (str_starts_with($value, self::COMPRESSION_MARKER) {
   $value = substr($value, strlen(COMPRESSION_MARKER));
   $value = gzuncompress($value);
}
```

Reply:
Uhhh, I like that idea! :-)

Reply:
So, we just need a marker if it's compressed 👍

Reply:
> So, we just need a marker if it's compressed

That was also my first idea but I added the explicit marker just for the case the actual request payload starts with the marker by chance.